src/redux/books/booksSlice.js

```diff
@@ -1,44 +1,88 @@
-    category: 'Fiction',
-  },
-  {
-    itemId: 'item2',
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-    category: 'Fiction',
-  },
-  {
-    itemId: 'item3',
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-    category: 'Nonfiction',
+import axios from 'axios';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const apiURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/0vHIXZ1klByIbEZsVgzM/books';
+
+export const getBookItems = createAsyncThunk('books/getBookItems', async (_, thunkAPI) => {
+  try {
+    const response = await axios.get(apiURL);
+    return response.data; // Return the fetched data as the payload
+  } catch (error) {
+    // throw error; // Rethrow the error to be caught by the rejected case
+    return thunkAPI.rejectWithValue('Request failed');
+  }
+});
+
+export const addBook = createAsyncThunk(
+  'books/addBook',
+  async (book, thunkAPI) => {
+    try {
+      const response = await axios.post(apiURL, book);
+      if (response.status === 201) {
+        thunkAPI.dispatch(getBookItems());
+      }
+      return null;
+    } catch (error) {
+      return thunkAPI.rejectWithValue('Request failed');
+    }
   },
-];
+);
+
+export const removeBook = createAsyncThunk('books/deleteBook',
+  async (itemId, thunkAPI) => {
+    try {
+      const response = await axios.delete(`${apiURL}/${itemId}`);
+
+      if (response.status === 201) {
+        thunkAPI.dispatch(getBookItems());
+      }
+      return null;
+    } catch (error) {
+      return thunkAPI.rejectWithValue('Book not deleted');
+    }
+  });
 
-export const booksSlice = createSlice({
+const booksSlice = createSlice({
   name: 'books',
-  initialState,
-  reducers: {
-    add: (state, action) => {
-      const {
-        title, author, category, itemId,
-      } = action.payload;
-      state.push({
-        itemId, title, author, category,
+  initialState: {
+    books: [],
+    isLoading: true,
+    error: null,
+  },
+
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(getBookItems.pending, (state) => {
+        state.isLoading = true;
+        state.error = null;
+      })
+      .addCase(getBookItems.fulfilled, (state, action) => {
+        console.log(action);
+        state.isLoading = false;
+        state.books = action.payload;
+      })
+      .addCase(getBookItems.rejected, (state) => {
+        state.isLoading = false;
+        state.error = true;
+      })
+      .addCase(addBook.pending, (state) => {
+        state.isLoading = true;
+        state.error = null;
+      })
+      .addCase(addBook.rejected, (state) => {
+        state.isLoading = false;
+        state.error = true;
+      })
+      .addCase(removeBook.pending, (state) => {
+        state.isLoading = true;
+        state.error = null;
+      })
+      .addCase(removeBook.rejected, (state) => {
+        state.isLoading = false;
+        state.error = true;
```

Comment:
-  You have done great so far :+1:  but there are a few changes you need to make before moving to the next milestone. :smile: :wave: 

- [ ] Looking at the screenshot below, you can see that there is an error in the console with your ` reducer(s) handling this action type: books/getBookItems/fulfilled, books/deleteBook/fulfilled.`. I'll suggest you fix as it might affect the correct functionality of the app.

- - Screenshot:
![Selection_015](https://github.com/pushkar1995/react-bookstore-app/assets/107632780/6eb3341d-a886-4c8b-b4e4-1499bd7e8d7d)